src/main/java/io/pinecone/configs/PineconeConfig.java

```diff
@@ -0,0 +1,63 @@
+package io.pinecone.configs;
+
+import io.grpc.ManagedChannel;
+import io.pinecone.exceptions.PineconeValidationException;
+
+public class PineconeConfig {
+
+    private String apiKey;
+    private String host;
+    private String usageContext;
+    private ManagedChannel customChannelBuilder;
+
+    public PineconeConfig(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public String getApiKey() {
+        return apiKey;
+    }
+
+    public void setApiKey(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    public void setHost(String host) {
+        this.host = host;
+    }
+
+    public String getUsageContext() {
+        return usageContext;
+    }
+
+    public void setUsageContext(String usageContext) {
+        this.usageContext = usageContext;
+    }
+
+    public ManagedChannel getCustomChannelBuilder() {
+        return this.customChannelBuilder;
+    }
+
+    public void setCustomChannelBuilder(ManagedChannel customChannelBuilder) {
+        this.customChannelBuilder = customChannelBuilder;
+    }
+
+    public interface CustomChannelBuilder {
+        ManagedChannel buildChannel();
+    }
+
+    void validate() {
+        if (apiKey == null)
+            throw new PineconeValidationException("Invalid PineconeConfig: missing apiKey");
+    }
+
+    public String getUserAgent() {
+        String userAgentLanguage = "lang=java; pineconeClientVersion = v0.8.0";
+        return (this.getUsageContext() != null) ?
+                userAgentLanguage + "; usageContext=" + this.getUsageContext() : userAgentLanguage;
```

Comment:
This is a nit but I find this hard to read, mainly because of the `;` inside the string concatenation being so visually similar to `:`. 

Would be nice for readability to write this out fully.

```suggestion
        if (this.getUsageContext() == null) {
            return userAgentLanguage;
        } else {
            return userAgentLanguage + "; usageContext=" + this.getUsageContext();
        }
```

Reply:
Sure, I can expand it.